hera/character/Portrait.tsx

```diff
@@ -30,43 +29,64 @@ export default memo(function Portrait({
   unit: UnitInfo;
   variant?: number;
 }) {
+  const ref = useRef<HTMLCanvasElement>(null);
   const hasPortraits = useSprites('portraits');
-  const sprite = hasPortraits
-    ? spriteURL('Portraits', encodeDynamicPlayerID(player))
-    : '';
 
   const { position } = unit.sprite.portrait;
-  const y = -(position.y + (variant || 0)) * PortraitHeight + 'px';
-  const alternateY = -(position.y + (variant || 0) + 6) * PortraitHeight + 'px';
+  const keyFrameYs = [
+    (position.y + (variant || 0)) * PortraitHeight,
+    (position.y + (variant || 0) + 6) * PortraitHeight,
+  ];
+  let curKeyFrameY = 0;
+
+  useLayoutEffect(() => {
+    if (!hasPortraits) {
+      return;
+    }
+
+    const canvas = ref.current;
+    if (!canvas) {
+      return;
+    }
+
+    const image = spriteImage('Portraits', encodeDynamicPlayerID(player));
+    const context = canvas.getContext('2d')!;
+
+    function animate() {
+      context.drawImage(
+        image,
+        position.x * PortraitWidth,
+        keyFrameYs[curKeyFrameY],
+        PortraitWidth,
+        PortraitHeight,
+        0,
+        0,
+        PortraitWidth,
+        PortraitHeight,
+      );
+
+      curKeyFrameY = (curKeyFrameY + 1) % keyFrameYs.length;
+    }
+
+    if (animate) {
+      setInterval(animate, 1000 / keyFrameYs.length);
+    }
```

Comment:
This needs to return () => clearInterval(interval) so that it gets stopped properly.

Reply:
Also the effect needs to support the `paused` flag and ensure the animation is not started.

Reply:
- returning () => clearInterval(interval)
[feat: cleanup useLayoutEffect](https://github.com/nkzw-tech/athena-crisis/pull/32/commits/a3f0f3b271008619c35d16434845c57a9db563ca)

- supporting `paused` flag
[feat: support paused flag](https://github.com/nkzw-tech/athena-crisis/pull/32/commits/ef766b5dca0a6497adc388f361d13fc69107c173)
[fix: render image when animate, paused are not given](https://github.com/nkzw-tech/athena-crisis/pull/32/commits/8b698e192170dfdb472c572484c28b2ef1ea70d6)